File: Behavioral/Iterator/Java/SongCollection.java
// SongCollection.java

import java.util.ArrayList;
import java.util.List;

public class SongCollection {
  private List<Song> songs = new ArrayList<>();

  public void addSong(Song song) {
    songs.add(song);
  }

  public Iterator<Song> iterator() {
    return new SongIterator(songs);
  }
}

File: Behavioral/Iterator/JavaScript/song_collection.js
// song_collection.js

const SongIterator = require("./song_iterator");

class SongCollection {
  constructor() {
    this.songs = [];
  }

  addSong(song) {
    this.songs.push(song);
  }

  getIterator() {
    return new SongIterator(this.songs);
  }
}

module.exports = SongCollection;
